robocorp-code/src/robocorp_code/inspector/java/robocorp_java/_event_pump.py

```diff
@@ -0,0 +1,67 @@
+import ctypes
+import platform
+import threading
+import time
+from concurrent import futures
+
+from JABWrapper.jab_wrapper import JavaAccessBridgeWrapper
+from robocorp_ls_core.robotframework_log import get_logger
+
+log = get_logger(__name__)
+
+
+REMOVE_FROM_QUEUE = 0x0001
+
+
+class EventPumpThread(threading.Thread):
+    def __init__(
+        self,
+    ) -> None:
+        super().__init__()
+        # Jab wrapper needs to be part of the thread that pumps the window events
+        self._jab_wrapper: JavaAccessBridgeWrapper = None
+        self._future: futures.Future = futures.Future()
+        self._quit_event_loop = threading.Event()
+
+    def _pump_background(self) -> bool:
+        try:
+            PeekMessage = ctypes.windll.user32.PeekMessageW  # type: ignore
+            TranslateMessage = ctypes.windll.user32.TranslateMessage  # type: ignore
+            DispatchMessage = ctypes.windll.user32.DispatchMessageW  # type: ignore
+
+            message = ctypes.byref(ctypes.wintypes.MSG())
+            # Nonblocking API to get windows window events from the queue.
+            # https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-peekmessagea
+            # Use non blocing API here so that the thread can quit.
+            if PeekMessage(message, 0, 0, 0, REMOVE_FROM_QUEUE):
+                TranslateMessage(message)
+                log.debug("Dispatching msg={}".format(repr(message)))
+                DispatchMessage(message)
+                return True
+        except Exception as err:
+            log.error(f"Pump error: {err}")
+        return False
+
+    def run(self) -> None:
+        if platform.system() != "Windows":
+            return
+
+        # Raise the error to the main thread from here
+        self._jab_wrapper = JavaAccessBridgeWrapper(ignore_callbacks=True)
+        self._future.set_result(self._jab_wrapper)
+        while not self._quit_event_loop.is_set():
+            # The pump is non blocking. If the is no message in the queue
+            # wait for 10 milliseconds until check again to prevent too
+            # fast loop.
+            # TODO: add backoff timer
+            if not self._pump_background():
+                time.sleep(0.1)
```

Comment:
I think this was raised (the comments still say 10 milliseconds).

As a note, usually the time I use if something must be "real-time" is `time.sleep(1/30)` (to have 30 updates/second), not sure the case here, but 10 updates per second as is now can be usually noticeable.

This needs an update to the comment (and my suggestion is 1/30, but if there's a reason for the higher value then just update the comment).